gms/gossiper.cc

```diff
@@ -917,19 +917,15 @@ void gossiper::run() {
                     logger.debug("Talk to live nodes: {}", live_nodes);
                     for (auto& ep: live_nodes) {
                         // Do it in the background.
-                        (void)do_gossip_to_live_member(message, ep).handle_exception([] (auto ep) {
-                            logger.trace("Failed to do_gossip_to_live_member: {}", ep);
-                        });
+                        (void)do_gossip_to_live_member(message, ep);
```

Comment:
What if the other place inside the function throws other than the send message function which has a handle_exception attached.

Reply:
Right now it doesn't. Even if it will _throw_ the .handle_exception() won't help, as it just captures exceptional futures, not throws

Reply:
Yes, this is one of the pitfalls. If we coroutinize the code, the throws will return exceptional futures. My point is that do_gossip_to_live_member can fail in multiple ways not just the send. It's better to keep the outer capture.

Reply:
This applies to pretty much every continuation chain -- if it starts to generate exceptions _later_ we'll lose those, so it's better to keep dead .handle_exception()s around. If someone will patch those functions so that they start to fail, it's _that_ patch duty to handle these cases, not current code's

Reply:
Then I do not see the point to remove it and add it back later. Please note, we do not wait for this fiber. The last thing we can do is to catch the exceptions.